### PR TITLE
fix: change defaultSpliceStrategy in array to use mergeSpliceStrategy, remove splice-strategy file

### PR DESCRIPTION
**Pull request by prudepixie**
# Pull Request

## 📖 Description

In fast-element 2.0, a new defaultSpliceStrategy was introduced, but this implementation introduced some bugs regarding calculation of where views were supposed to be correctly added or deleted according to the itemsBinding, it resulted in an array of Splices that mirrored each array operation. 
The mergeSpliceStrategy from fast-element 1.0 solves this problem by making all the calculations beforehand and notifying a single Splice to be passed to the Observer.  
Some implementation details here required a bit of the combination of code from defaultSpliceStrategy and mergeSpliceStrategy. Specially to handle reverse() and sort() array methods, these 2 methods do not 'add Splices' like other array modification methods. Instead, we want to reset when met with these 2 methods. Meaning there needed to be a slight modification for the defaultSpliceStrategy.normalize code.

With this change, we no longer need the splice-strategies file, so that is removed. 

X (UPDATE: this breaks scenarios, will not add) 
_Also, since this change will only resolve in 1 Splice (no longer an array of Splices), the Repeat directive updateViews code is changed to reflect this change, where we no longer need to loop through an array of Splices and store Views to be reused, but only doing this operation once._ 

- change defaultStrategies back to mergeSpliceStrategy (from fast-element-1)
- move logic from splice-strategies.ts to array.ts
- delete splice-strategies.ts, remove reference from package.json
- add back to back scenario tests
- X (UPDATE: this breaks scenarios, will not add) _update repeat directive updateViews() to take in 1 Splice (not an array of Splices)_

### 🎫 Issues
#6177 

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-element/src/templating/repeat.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ export class RepeatBehavior<TSource = any> implements Behavior, Subscriber {
         } else if (args[0].reset) {
             this.refreshAllViews();
         } else {
-            this.updateViews(args as Splice[]);
+            this.updateViews(args[0] as Splice);
```
Comment (EisenbergEffect) [inline packages/web-components/fast-element/src/templating/repeat.ts:152]: This is the one thing I'm not sure about. It seems that if there are multiple changes, that the `project` code can't necessarily merge those down to a single splice. There still might be multiple splices. We might need some tests that mix up a wider variety of array mutations to see this issue.

**Review by EisenbergEffect — COMMENTED** [review]
I want to investigate further the change in `updateViews`. I'm not sure that we can count on there only being one splice. I haven't looked at the code in a while, as you know. So, I could be wrong. Let's double check though.

File: packages/web-components/fast-element/src/templating/repeat.ts (hunk 1 above)
Comment (prudepixie) [reply packages/web-components/fast-element/src/templating/repeat.ts]: Hey, good catch, after adding some combinations of array modification tests, this indeed needed to be reverted back to handle array of Splices.

**Comment by EisenbergEffect** [comment]
Where are we at on this? Does it need a refreshed review?

**Comment by prudepixie** [comment]
> Where are we at on this? Does it need a refreshed review?

Yes, can you do a quick review on the Repeat.ts, there's just one line of code that was added after the discussions

**Review by EisenbergEffect — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

**Pull request merged**